### ISSUE TITLE
How to configure ossindex-maven to use a proxy ?

### ISSUE THREAD

**Issue by farnulfo** [sonatype/ossindex-maven#73]
I tried 
`
<plugin>

          <groupId>org.sonatype.ossindex.maven</groupId>
          <artifactId>ossindex-maven-plugin</artifactId>
          <configuration>
            <clientConfiguration>
              <proxy>
                <host>localhost</host>
                <port>8080</port>
              </proxy>
            </clientConfiguration>
          </configuration>
          <executions>
            <execution>
              <id>audit-dependencies</id>
              <phase>validate</phase>
              <goals>
                <goal>audit</goal>
              </goals>
            </execution>
          </executions>
        </plugin>        
`
=>
Unable to parse configuration of mojo org.sonatype.ossindex.maven:ossindex-maven-plugin:3.2.0:audit for parameter proxy: Cannot find 'proxy' in class org.sonatype.ossindex.service.client.OssindexClientConfiguration

**Comment by farnulfo** [comment]
I found the right configuration:
```
<plugin>
    <groupId>org.sonatype.ossindex.maven</groupId>
    <artifactId>ossindex-maven-plugin</artifactId>
    <configuration>
        <clientConfiguration>
            <proxyConfiguration>
                <host>host</host>
                <port>8080</port>
                <authConfiguration>
                    <username>xxxxxxxx</username>
                    <password>yyyyyyyyy</password>
                </authConfiguration>
            </proxyConfiguration>
        </clientConfiguration>
    </configuration>
    <executions>
        <execution>
            <id>audit-dependencies</id>
            <phase>validate</phase>
            <goals>
                <goal>audit</goal>
            </goals>
        </execution>
    </executions>
</plugin>
```

I started with https://sonatype.github.io/ossindex-maven/maven-plugin/ossindex-audit/#clientConfiguration
There is a 404 for the "client configuration" link https://sonatype.github.io/ossindex-public/maven/apidocs/org/sonatype/ossindex/service/client/OssindexClientConfiguration.html but I used it to search and find the source code:
https://github.com/sonatype/ossindex-public/blob/main/client/src/main/java/org/sonatype/ossindex/service/client/OssindexClientConfiguration.java .
With this source code I've found the right xml tag based on it.

**Comment by sostrich** [comment]
Hi @farnulfo, thanks so much for the explanation, that really helped me get a bit further.

One question are you sure that clientConfiguration has to be nested inside proxyConfiguration? I managed to make it work with

```
configuration>
        <clientConfiguration>
              <authConfiguration>
                  <username>xxxxxxxx</username>
                  <password>yyyyyyyyy</password>
              </authConfiguration>
        </clientConfiguration>
``` 
(i.e. when I added the wrong username here, i got a 401.

I'm trying to get it to work from the command line, sadly 
`mvn org.sonatype.ossindex.maven:ossindex-maven-plugin:audit-aggregate -f pom.xml -"Dossindex.clientConfiguration.authConfiguration.username={WRONG_USERNAME}" -"Dossindex.clientConfiguration.authConfiguration.password={SOME_PASSWORD}"` still does not return a 401, but the correct cve result, so I think this is not how you run it from the command line
I find that the documentation is really lacking here.:/
